contracts/integrations/test/internal-integration-tests/exchange_wrapper_test.ts

```diff
@@ -0,0 +1,1058 @@
+        });
+
+        it('should transfer the correct amounts, pay the protocol fee in eth, and refund for excess eth sent', async () => {
+            await testFillOrKillOrderAsync(DeploymentManager.protocolFee.times(2));
+        });
+
+        it('should transfer the correct amounts and pay the protocol fee in weth', async () => {
+            await testFillOrKillOrderAsync(0);
+        });
+
+        it('should transfer the correct amounts, pay the protocol fee in weth, and refund for excess eth sent', async () => {
+            await testFillOrKillOrderAsync(DeploymentManager.protocolFee.div(2));
+        });
+
+        it('should revert if a signedOrder is expired', async () => {
+            const currentTimestamp = await getLatestBlockTimestampAsync();
+            const signedOrder = await makerActor.signOrderAsync({
+                expirationTimeSeconds: new BigNumber(currentTimestamp).minus(10),
+            });
+            const orderHashHex = orderHashUtils.getOrderHashHex(signedOrder);
+            const expectedError = new ExchangeRevertErrors.OrderStatusError(orderHashHex, OrderStatus.Expired);
+            const tx = deployment.exchange.fillOrKillOrder.awaitTransactionSuccessAsync(
+                signedOrder,
+                signedOrder.takerAssetAmount,
+                signedOrder.signature,
+                { from: taker, gasPrice: DeploymentManager.gasPrice, value: DeploymentManager.protocolFee },
+            );
+            return expect(tx).to.revertWith(expectedError);
+        });
+
+        it('should revert if entire takerAssetFillAmount not filled', async () => {
+            const signedOrder = await makerActor.signOrderAsync();
+            const takerAssetFillAmount = signedOrder.takerAssetAmount;
+
+            await deployment.exchange.fillOrder.awaitTransactionSuccessAsync(
+                signedOrder,
+                signedOrder.takerAssetAmount.dividedToIntegerBy(2),
+                signedOrder.signature,
+                { from: taker, gasPrice: DeploymentManager.gasPrice, value: DeploymentManager.protocolFee },
+            );
+            const expectedError = new ExchangeRevertErrors.IncompleteFillError(
+                ExchangeRevertErrors.IncompleteFillErrorCode.IncompleteFillOrder,
+                takerAssetFillAmount,
+                takerAssetFillAmount.dividedToIntegerBy(2),
+            );
+            const tx = deployment.exchange.fillOrKillOrder.awaitTransactionSuccessAsync(
+                signedOrder,
+                signedOrder.takerAssetAmount,
+                signedOrder.signature,
+                { from: taker, gasPrice: DeploymentManager.gasPrice, value: DeploymentManager.protocolFee },
+            );
+            return expect(tx).to.revertWith(expectedError);
+        });
+    });
+
+    describe('batch functions', () => {
+        let signedOrders: SignedOrder[];
+        beforeEach(async () => {
```

Comment:
can we make this a `before(` ? Since the entire test suite is wrapped by `blockchainTests.resets(`?